mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java

```diff
@@ -352,18 +354,26 @@ public Future<List<Record>> getMatchedRecordsWithoutIndexersVersionUsage(MatchFi
 
   private Condition getMatchedFieldCondition(MatchField matchedField, String partition) {
     Map<String, String> params = new HashMap<>();
+    var qualifierSearch = false;
     params.put("partition", partition);
     params.put("value", getValueInSqlFormat(matchedField.getValue()));
+    if (matchedField.getQualifierMatch() != null) {
+      qualifierSearch = true;
+      params.put("qualifier", getSqlQualifier(matchedField.getQualifierMatch()));
+    }
+    String sql;
+    Condition condition;
```

Comment:
if condition is not used, let's remove it

Reply:
These conditions are used.
`(matchedField.getQualifierMatch() != null` is used to keep the logic that was before we added the search by qualifiers unchanged
`if (matchedField.isControlField())` this condition was there before, because if it is a control field, the sql query should be formed differently